mrmustard/math/tensorflow.py

```diff
@@ -336,23 +349,23 @@ def grad(dLdpoly):
 
         return poly, grad
 
-    def eigvals(self, tensor: tf.Tensor) -> Tensor:
+    def eigvals(tensor: tf.Tensor) -> Tensor:
```

Comment:
Not sure you can just remove `self` given that this is a method of the `TFMath` class. Maybe what you want to do is decorate it with `@staticmethod`?

Reply:
hmm I think Zhi wrote these. Looks like he forgot the self argument in a few of them?

Reply:
Not sure it was "forgotten" but rather "removed" (perhaps on the advice of `pylint` because the method doesn't actually use it?). Still, I believe the tests are currently failing because of this change, and would recommend either putting it back or making these `@staticmethod`s.

Reply:
Thanks @heltluke! I indeed removed `self` following pylint suggestions.